### ISSUE TITLE
Messages not being sent to/received by Azure Insights

### ISSUE THREAD

**Issue by kf6kjg** [ovhemert/pino-applicationinsights#42]
## 🐛 Bug Report

The messages sent to Applications Insights do not appear to be being received.

## To Reproduce

```sh
$ echo '{"level":50,"time":1617656797803,"pid":82,"hostname":"fd712adddba1","msg":"Test at level 50."}' | npx pino-applicationinsights --key REDACTED
logging

```

The REDACTED value in the above comes from the Application Insights Instrumentation Key line, the location of which on the Azure Portal interface is highlighted in green here:
![image](https://user-images.githubusercontent.com/1157224/113630636-0a8a8400-961d-11eb-8d24-2683228050c0.png)

Open an additional terminal and run the following

```sh
az login
az monitor app-insights query --app REDACTED --analytics-query 'traces'
```

The REDACTED value here is the `Application ID` located under the `API Access` sidebar. 

Alternatively, use the `Logs` sidebar and query for traces.

## Expected behavior

The logger to send the log object to the Azure Insights service and then the latter to be able to show the resulting entry.

## Your Environment

- *os*: Linux: Alpine 3.11 via the `node:14-alpine3.11` Docker image.

**Comment by kf6kjg** [comment]
According to the [Azure documentation](https://docs.microsoft.com/en-us/azure/azure-monitor/app/create-new-resource#copy-the-instrumentation-key) it looks like the instrumentation keys are being phased out in favor of connection strings.  It's possible that this is the source of the disconnect?

**Comment by gboston** [comment]
@kf6kjg Any fix for this?

**Comment by kf6kjg** [comment]
> @kf6kjg Any fix for this?

None that I'm aware of at this time.

**Comment by kf6kjg** [comment]
AFAICT this is still a problem, though I've not tried reproducing it since I reported it.

**Comment by devinrhode2** [comment]
I recently got application insights logging working just fine, using next.js and next-logger, pino-ms, and pino-applicationinsights.

I am using the connection string in place of the instrumentation key. The MS docs don't make it very clear. But it seems anywhere you would use an instrumentation key ("iKey") you can use a connection string. 

https://github.com/atkinchris/next-logger/issues/12

In next.config.js, if `process.env.APPINSIGHTS_CONNECTIONSTRING` is not defined, I recommend crashing so a build+deploy does not go through. Unfortunately, why that env var would not come through is an azure issue, has nothing to do with pino-applicationinsights. But if you have that issue, I'm not opposed to comparing your pipeline file to ours.

**Comment by devinrhode2** [comment]
As I've had this working, I recommend closing the issue. Maybe the issue was solved by some other related dependency, maybe INSTRUMENTATION_KEY just does not work anymore, idk.

**Comment by kf6kjg** [comment]
If that be the case, then maybe a documentation update is all that is needful.

**Comment by devinrhode2** [comment]
I seem to have been mistaken in my previous analysis. In azure portal under "Diagnose and Solve Problems" it shows "Application Logs" (stdout) from my node server, without any of my logging code.

**Comment by kf6kjg** [comment]
Application Insights != stdout. I’d expect nothing on stdout.

**Comment by devinrhode2** [comment]
Yeah I noticed when ApplicationInsights setup code runs, there isn't anything printed to stdout except this:
```
ApplicationInsights:An invalid instrumentation key was provided. There may be resulting telemetry loss [ 'asdf' ]
```
